flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -301,6 +306,14 @@ public void close() throws Exception {
                 exception = ExceptionUtils.firstOrSuppressed(e, exception);
             }
 
+            final File workingDirectory = ClusterEntrypointUtils.getWorkingDirectory(configuration);
+
+            try {
+                FileUtils.deleteDirectory(workingDirectory);
+            } catch (IOException ioe) {
+                exception = ExceptionUtils.firstOrSuppressed(ioe, exception);
+            }
+
```

Comment:
Why are we cleaning up the WD when closing the task manager? Shouldn't we keep it in case the termination is on purpose?

Reply:
I think we need to distinguish between a graceful shutdown and a non-graceful shutdown. In case of a SIGTERM/SIGKILL we should not clean up. However, if the user said to shut down, then we should clean up the resources.

Reply:
At the moment, I think the `TaskManagerRunner` is not reacting to process signals.

Reply:
But we are when seeing a fatal error. I think this is not right.

Reply:
> However, if the user said to shut down, then we should clean up the resources.

But what if the user wants to migrate the process to another machine (e.g. VM migration) while keeping the state? E.g. I wanna migrate this task manager to a more powerful node, so I SIGTERM it to stop it and then restart the task manager in another VM, while mounting the same Persistent volume

Reply:
SIGTERM I would not consider as a graceful shut down.

Reply:
I've pushed some fix up commits to fix the problem on TM side. I will do the same for the JM side now.

Reply:
I've pushed another fixup that addresses the clean up of the working directory on the `ClusterEntrypoint` side. I will add tests for this part when opening a PR for making the `ResourceID` configurable on the `ClusterEntrypoint` side.